### PR TITLE
Add thermal pad support for SO/Quad landpatterns

### PR DESCRIPTION
**Pull request by m-hilgendorf**

### CODE REVIEW

File: tests/test-land-patterns.stanza [hunk 1]
```diff
@@ -14,50 +14,74 @@ defpackage ocdb/test/land-patterns:
   import ocdb/symbols
   import ocdb/land-protrusions
 
-pcb-component QFP:
-  pcb-landpattern LQFP48:
-    make-qfp-landpattern(48,   ; number of pins 
+pcb-module soic-lands:
+  pcb-landpattern SO8N (thermal?:True|False):
+    if not thermal?:
```
Comment (bpellenard) [inline tests/test-land-patterns.stanza:19]:
```
val shape = Rectangle(3.0, 4.8) when thermal? else false
make-n-pin-soic-landpattern(8, 1.27, tol(6.0, 0.2), tol(3.9, 0.1), tol(4.9, 0.1), min-typ-max(0.4, 1.04, 1.27), min-typ-max(0.28, 0.38, 0.48), shape)
```

File: tests/test-land-patterns.stanza [hunk 2]
```diff
@@ -14,50 +14,74 @@ defpackage ocdb/test/land-patterns:
...
+    make-box-symbol()
+  
+  inst u1: soic-component(true)
+  inst u2: soic-component(false)
+  place(u1) on Top
+  place(u2) on Top
+
+pcb-module quad-lands:
+  pcb-landpattern LQFP48 (thermal?:True|False):
+    if not thermal?:
```
Comment (bpellenard) [inline tests/test-land-patterns.stanza:40]: likewise

File: utils/land-patterns.stanza [hunk 3]
```diff
@@ -1188,56 +1203,194 @@ public defn make-quad-landpattern (num-pins-west: Int,  ; the number of pins on
...
+                                   pitch-east:Double,   ; the pitch on the east face
+                                   pitch-north:Double,  ; the pitch on the north face
+                                   component-length: Toleranced, ; the distance from pad-edge to pad-edge on the exterior of the part in the x direction
+                                   component-width: Toleranced, ; the distance from pad-edge to pad-edge on the exterior of the part in the y direction
+                                   package-length: Toleranced,  ; the size of the package in the x direction
+                                   package-width: Toleranced,  ; the size of hte package in the y direction
+                                   lead-length: Toleranced, ; the dimension of the leads in the x direction on the east/west faces
+                                   lead-width: Toleranced, ; the dimension of the leads in the y direction on the east/west faces
+                                   lead-type:  LandProtrusionType, ; the type of the leads
+                                   exposed-metal-heat-feature?: Shape|False, ; the thermal pad
```
Comment (bpellenard) [inline utils/land-patterns.stanza:1276]: remove comma

File: utils/land-patterns.stanza [hunk 4]
```diff
@@ -1188,56 +1203,194 @@ public defn make-quad-landpattern (num-pins-west: Int,  ; the number of pins on
...
+
+; Create a quad landpattern with optional thermal pad
 public defn make-quad-landpattern (num-pins:Int, 
                                    pitch:Double, 
                                    component-size:Toleranced, 
                                    package-size:Toleranced, 
                                    lead-length:Toleranced, 
                                    lead-width:Toleranced, 
-                                   lead-type:LandProtrusionType):
+                                   lead-type:LandProtrusionType
```
Comment (bpellenard) [inline utils/land-patterns.stanza:1293]: missing comma

File: utils/land-patterns.stanza [hunk 5]
```diff
@@ -1188,56 +1203,194 @@ public defn make-quad-landpattern (num-pins-west: Int,  ; the number of pins on
...
+                                   lead-length:Toleranced, 
+                                   lead-width:Toleranced, 
+                                   lead-type:LandProtrusionType):
+  make-quad-landpattern(num-pins,
+                        pitch,
+                        component-size,
+                        package-size,
+                        lead-length,
+                        lead-width,
+                        lead-type
```
[the remark this answers is not in the document]
Comment (m-hilgendorf) [reply utils/land-patterns.stanza:1324]: fixed

File: utils/land-patterns.stanza (hunk 4 above)
Comment (m-hilgendorf) [reply utils/land-patterns.stanza:1300]: fixed

File: utils/land-patterns.stanza (hunk 3 above)
Comment (m-hilgendorf) [reply utils/land-patterns.stanza:1283]: fixed

File: tests/test-land-patterns.stanza (hunk 2 above)
Comment (m-hilgendorf) [reply tests/test-land-patterns.stanza]: done, thanks!

**Review by d-haldane — APPROVED** [review]

**Pull request merged**